editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.tsx

```diff
@@ -0,0 +1,198 @@
+  features,
+}) => {
+  const { schema, activeIndex, formik } = useMapAndLabelContext();
+  const [activeTab, setActiveTab] = useState<string>(
+    features[features.length - 1].properties?.label || "",
+  );
+
+  const handleChange = (_event: React.SyntheticEvent, newValue: string) => {
+    setActiveTab(newValue);
+  };
+
+  return (
+    <Box
+      sx={{
+        flexGrow: 1,
+        bgcolor: "background.paper",
+        display: "flex",
+        maxHeight: "fit-content",
+      }}
+    >
+      <TabContext value={activeTab}>
+        <Tabs
+          orientation="vertical"
+          variant="scrollable"
+          value={activeTab}
+          onChange={handleChange}
+          aria-label="Vertical tabs example"
+          sx={{ borderRight: 1, borderColor: "divider" }}
+        >
+          {features.map((feature, i) => (
+            <Tab
+              key={`tab-${i}`}
+              value={feature.properties?.label}
+              label={`${schema.type} ${feature.properties?.label}`}
+              {...a11yProps(i)}
+            />
+          ))}
+        </Tabs>
+        {features.map((feature, i) => (
+          <TabPanel
+            key={`tabpanel-${i}`}
+            value={feature.properties?.label}
+            sx={{ width: "100%" }}
+          >
+            <Typography component="h2" variant="h3">
+              {`${schema.type} ${feature.properties?.label}`}
+            </Typography>
+            <Typography variant="body2" gutterBottom>
+              {`${feature.geometry.type}`}
+              {feature.geometry.type === "Point"
+                ? ` (${feature.geometry.coordinates.map((coord) =>
+                    coord.toFixed(5),
+                  )})`
+                : ` (area ${
+                    feature.properties?.["area.squareMetres"] || 0
+                  } m²)`}
+            </Typography>
+            <SchemaFields
+              schema={schema}
+              activeIndex={activeIndex}
+              formik={formik}
+            />
+          </TabPanel>
+        ))}
+      </TabContext>
+    </Box>
+  );
+};
+
+const Root = () => {
+  const { validateAndSubmitForm, mapAndLabelProps } = useMapAndLabelContext();
+  const {
+    title,
+    description,
+    info,
+    policyRef,
+    howMeasured,
+    drawColor,
+    drawType,
+    schemaName,
+  } = mapAndLabelProps;
+
+  const teamSettings = useStore.getState().teamSettings;
+  const passport = useStore((state) => state.computePassport());
+
+  const [features, setFeatures] = useState<Feature[] | undefined>(undefined);
+
+  useEffect(() => {
+    const geojsonChangeHandler = ({ detail: geojson }: any) => {
+      if (geojson["EPSG:3857"]?.features) {
+        setFeatures(geojson["EPSG:3857"].features);
+      } else {
+        // if the user clicks 'reset' on the map, geojson will be empty object, so set features to undefined
+        setFeatures(undefined);
+      }
+    };
+
+    const map: any = document.getElementById("map-and-label-map");
```

Comment:
```suggestion
    const map: HTMLDivElement | null = document.getElementById("map-and-label-map");
```

Reply:
And then maybe throw an exception if no map found in order to type narrow within this function?